### ISSUE TITLE
RegexpMultilineCheck: Add option to specify which capturing group determines violation position.

### ISSUE THREAD

**Issue by Hariom-kr** [checkstyle/checkstyle#18812]
<html>

I have read check documentation: https://checkstyle.org/checks/regexp/regexpmultiline.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words</p>

When using RegexpMultilineCheck for conditional pattern matching, users must include context in the regex pattern. Currently, violations are reported at the start of the entire match (group 0), which reports the context line instead of the actual violation line.

<pre><code class="language-bash">Windows@Windows MINGW64 ~/check
$ cat Example.java
class Example {
    void method1() {
        // CONTEXT
        System.out.print("Should be reported here");
    }

    void method2() {
        System.out.print("This alone is OK");
    }
}
</code></pre>
<pre><code class="language-xml">Windows@Windows MINGW64 ~/check
$ cat config.xml
&lt;?xml version="1.0"?&gt;
&lt;!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd"&gt;
&lt;module name="Checker"&gt;
  &lt;module name="RegexpMultiline"&gt;
    &lt;property name="matchAcrossLines" value="true"/&gt;
    &lt;property name="format" value="(// CONTEXT\n\s*)(System\.out\.print\()"/&gt;
    &lt;property name="message" value="Console print after specific context"/&gt;
  &lt;/module&gt;
&lt;/module&gt;
</code></pre>
<pre><code class="language-bash">Windows@Windows MINGW64 ~/check
$ java -jar checkstyle-13.1.0-all.jar -c config.xml Example.java
Starting audit...
[ERROR] C:\Users\hario\check\Example.java:3: Console print after specific context [RegexpMultiline]
Audit done.
Checkstyle ends with 1 errors.
</code></pre>
<p><strong>Current Behavior:</strong>
Violation is reported on <strong>line 3</strong> (<code>// CONTEXT</code> comment), but the actual <code>System.out.print()</code> is on <strong>line 4</strong>.</p>
<p><strong>Expected Behavior:</strong>
With a new <code>reportGroup</code> property:</p>
<pre><code class="language-xml">&lt;module name="Checker"&gt;
  &lt;module name="RegexpMultiline"&gt;
    &lt;property name="matchAcrossLines" value="true"/&gt;
    &lt;property name="format" value="(// CONTEXT\n\s*)(System\.out\.print\()"/&gt;
    &lt;property name="reportGroup" value="2"/&gt;
    &lt;property name="message" value="Console print after specific context"/&gt;
  &lt;/module&gt;
&lt;/module&gt;
</code></pre>
<p>Expected output:</p>
<pre><code class="language-bash">Starting audit...
[ERROR] C:\Users\hario\check\Example.java:4: Console print after specific context [RegexpMultiline]
Audit done.
Checkstyle ends with 1 errors.
</code></pre>
<p>Violation correctly reported on <strong>line 4</strong> where <code>System.out.print()</code> actually is.</p>
<p><strong>Capturing group breakdown:</strong></p>

Group | Captures | Line
-- | -- | --
0 | Entire match | Line 3
1 | // CONTEXT\n | Line 3
2 | System.out.print( | Line 4

<p><strong>Why this is needed:</strong>
Java regex does not support variable-length lookbehinds, so this is not possible:</p>
<pre><code>(?&lt;=// CONTEXT\n\s*)System\.out\.print\(
</code></pre>
<p>Users are forced to include context as part of the match, causing violations to report at the wrong line.</p>
</html>

**Comment by romani** [comment]
What is practical reason of this? Real use case
We will have to put example in documentation, example should be real example

**Comment by romani** [comment]
I think it will be required with specific message, as in message user might reference to specific group/part of match. So this property will help message to be targeted and more clear to point to exact line.


-----

Enforcing Security Best Practices (e.g., Logging)
You might want to ensure that Password objects are never passed directly into a logger.

The Regex: `(LOG\.(?:info|debug|error)\(\s*.*)(password)(.*\))`

The Scenario:
```
LOG.error("Login failed for user: " + // Line 85
          user.getName() +            // Line 86
          " with password: " +        // Line 87
          password);                  // Line 88

```

------

Preventing "Magic Numbers" in Specific Contexts
Imagine a project where you are allowed to use "magic numbers" (like 100) generally, but never as a timeout value in a specific library call.

The Regex: `(setConnectionTimeout\(\s*)([0-9]+)(\s*\))`

The Scenario: The method call might span multiple lines:
```
client.setConnectionTimeout( // Line 40
    1000                     // Line 41
);

```

The Utility: If you flag the whole match, the error appears on Line 40. By setting reportGroup to 2, the error correctly highlights Line 41, where the actual hardcoded number is located.

**Comment by ishasingh1505** [comment]
I'm on it.

**Comment by romani** [comment]
@Hariom-kr, next time please do comment, I am on it.
To avoid concurrency.
